io/net/infiniband/mckey.py

```diff
@@ -72,12 +72,23 @@ def setUp(self):
                                              default="None")
         self.ipaddr = self.params.get("host_ip", default="")
         self.netmask = self.params.get("netmask", default="")
+        local = LocalHost()
         if self.iface[0:2] == 'ib':
-            configure_network.set_ip(self.ipaddr, self.netmask, self.iface,
-                                     interface_type='Infiniband')
+            self.networkinterface = NetworkInterface(self.iface, local,
+                                                     if_type='Infiniband')
+            try:
+                self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            except:
+                self.networkinterface.save(self.ipaddr, self.netmask)
         else:
-            configure_network.set_ip(self.ipaddr, self.netmask, self.iface,
-                                     interface_type='Ethernet')
+            self.networkinterface = NetworkInterface(self.iface, local)
+            try:
+                self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            except:
+                self.networkinterface.save(self.ipaddr, self.netmask)
+            self.networkinterface.bring_up()
```

Comment:
This bringup should be outside of if condition right?  If not it works only for ROCE and not for IB

Reply:
@manvanthar  yes, i will change that.